tracer/src/Datadog.Trace/Agent/MessagePack/TraceChunkModel.cs

```diff
@@ -62,13 +67,13 @@ public TraceChunkModel(in ArraySegment<Span> spans)
     }
 
     // used only to chain constructors
-    private TraceChunkModel(in ArraySegment<Span> spans, TraceContext? traceContext)
+    private TraceChunkModel(in ArraySegment<Span> spans, TraceContext? traceContext, int? samplingPriority)
         : this(spans, traceContext?.RootSpan)
     {
         if (traceContext is not null)
```

Comment:
I'm not sure if it makes too much of a difference, but maybe before the if statement you could assign `SamplingPriority = samplingPriority` and then inside the if statement you do `SamplingPriority ??= traceContext.SamplingPriority`?